### ISSUE TITLE
PSNR: Nan

### ISSUE THREAD

**Issue by YuhangLiu98** [swz30/Restormer#121]
![Image](https://github.com/user-attachments/assets/7654b7b8-3344-442a-aa13-401b1a926b39)

I've run it on multiple other datasets and this is happening and I'm very confused.

**Comment by Mytttttttt** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by HuaqingHe** [comment]
I'm encountering a similar issue in my experiments and would like to respectfully seek your guidance. My current configuration is as follows:

Dataset: DIV2K + FFHQ
train:
  ema_decay: 0
  optim_g:
    type: Adam
    lr: 4e-4
    weight_decay: 0
    betas: [0.9, 0.99]
  
  scheduler:
    type: CosineAnnealingRestartLR
    periods: [200000]
    restart_weights: [1]
    eta_min: 1e-7
pretrain_network_g: experiments/pretrained_models/Restormer_real_denoising.pth
Notably, I'm observing the [specific issue you're referring to] as early as Step 2000. Could you kindly advise on potential reasons for this early occurrence and any recommended adjustments to the configuration?
Also, I have changed the scheduler to
type: MultiStepLR
    milestones: [100000, 750000]
    #milestones: [5000, 10000, 15000, 20000，25000，30000，35000]
    gamma: 0.5
The problem is still.

**Comment by HuaqingHe** [comment]
> ![Image](https://github.com/user-attachments/assets/7654b7b8-3344-442a-aa13-401b1a926b39)
> 
> I've run it on multiple other datasets and this is happening and I'm very confused.

I find that `use_grad_clip` is helpful. Did you turn it on?

**Comment by YuhangLiu98** [comment]
> > ![Image](https://github.com/user-attachments/assets/7654b7b8-3344-442a-aa13-401b1a926b39)
> > I've run it on multiple other datasets and this is happening and I'm very confused.
> 
> I find that `use_grad_clip` is helpful. Did you turn it on?

在yml文件中，use_grad_clip: true。需要将它设置为False吗？

**Comment by HuaqingHe** [comment]
可以看看 model 的optimize_parameters里面这个参数有没有被用到。是要打开的。
`if self.opt['train'].get('use_grad_clip', False):
            torch.nn.utils.clip_grad_norm_(self.net_g.parameters(), 0.01)`

打开之后会限制每次梯度更新不超过0.01。如果还是有问题，可以试试把learning rate调小。
我试着可以把 0.01 调大点。不然每次更细的梯度太小，模型参数变化不大，收敛很慢。



---Original---
From: "Yuhang ***@***.***&gt;
Date: Tue, Apr 22, 2025 17:07 PM
To: ***@***.***&gt;;
Cc: "Huaqing ***@***.******@***.***&gt;;
Subject: Re: [swz30/Restormer] PSNR: Nan (Issue #121)


     
image.png (view on web)
 I've run it on multiple other datasets and this is happening and I'm very confused.
  
I find that use_grad_clip is helpful. Did you turn it on?
  
在yml文件中，use_grad_clip: true。需要将它设置为False吗？

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  YuhangLiu98 left a comment (swz30/Restormer#121)
   
image.png (view on web)
 I've run it on multiple other datasets and this is happening and I'm very confused.
  
I find that use_grad_clip is helpful. Did you turn it on?
  
在yml文件中，use_grad_clip: true。需要将它设置为False吗？
